src/analysis/normed_space/deriv.lean

```diff
@@ -284,6 +285,77 @@ hf.comp (hg.mono hf.continuous_at)
 
 end
 
+section
+
+variables {K : Type*} [nondiscrete_normed_field K]
+variables {E : Type*} [normed_space K E]
+variables {F : Type*} [normed_space K F]
+include K
+
+open topological_space
+
+/-- The differential of a map at a point along a filter is unique, given that filter is coarser than the
+ neighbourhood filter of the point.-/
+lemma fderiv_at_filter_unique (f : E → F) (x₀ : E) {L : filter E} (h : nhds x₀ ≤ L) {A₁ A₂ : E → F} :
```

Comment:
Is there a way to make sure that the beginning and end of the proof are clearly identified? You are following mathlib style, sure, but I feel it is confusing. Maybe avoid blank lines in the middle of the proof (just adding `--` on a blank line would help), and adding `--end of proof` at the end?